app/models/mention.rb

```diff
@@ -5,16 +5,22 @@
 class Mention < ActiveRecord::Base
   REGEX = /@\{([^;]+); ([^\}]+)\}/
 
-  belongs_to :post
+  belongs_to :mentions_container, polymorphic: :true
+  belongs_to :post, ->(mention) { (mention.mentions_container_type == "Post") ? where("1 = 1") : none },
```

Comment:
Ok, I'll check it once again then. One of my purposes for that was to escape rewriting all the code which references to Mention through post.